connectors/camel-sjms2-kafka-connector/src/main/java/org/apache/camel/kafkaconnector/sjms2/CamelSjms2SourceConnectorConfig.java

```diff
@@ -0,0 +1,237 @@
+    public static final String CAMEL_SOURCE_SJMS2_ENDPOINT_INCLUDE_ALL_JMSX_PROPERTIES_DOC = "Whether to include all JMSXxxx properties when mapping from JMS to Camel Message. Setting this to true will include properties such as JMSXAppID, and JMSXUserID etc. Note: If you are using a custom headerFilterStrategy then this option does not apply.";
+
+    public static final Boolean CAMEL_SOURCE_SJMS2_ENDPOINT_MAP_JMS_MESSAGE_DEFAULT = true;
+    public static final String CAMEL_SOURCE_SJMS2_ENDPOINT_MAP_JMS_MESSAGE_CONF = "camel.source.endpoint.mapJmsMessage";
+    public static final String CAMEL_SOURCE_SJMS2_ENDPOINT_MAP_JMS_MESSAGE_DOC = "Specifies whether Camel should auto map the received JMS message to a suited payload type, such as javax.jms.TextMessage to a String etc. See section about how mapping works below for more details.";
+
+    public static final Boolean CAMEL_SOURCE_SJMS2_ENDPOINT_SHARED_JMS_SESSION_DEFAULT = true;
+    public static final String CAMEL_SOURCE_SJMS2_ENDPOINT_SHARED_JMS_SESSION_CONF = "camel.source.endpoint.sharedJMSSession";
+    public static final String CAMEL_SOURCE_SJMS2_ENDPOINT_SHARED_JMS_SESSION_DOC = "Specifies whether to share JMS session with other SJMS endpoints. Turn this off if your route is accessing to multiple JMS providers. If you need transaction against multiple JMS providers, use jms component to leverage XA transaction.";
+
+    public static final String CAMEL_SOURCE_SJMS2_PATH_DESTINATION_TYPE_DEFAULT = "queue";
+    public static final String CAMEL_SOURCE_SJMS2_PATH_DESTINATION_TYPE_CONF = "camel.source.path.destinationType";
+    public static final String CAMEL_SOURCE_SJMS2_PATH_DESTINATION_TYPE_DOC = "The kind of destination to use: queue or topic.";
+
+    public static final String CAMEL_SOURCE_SJMS2_PATH_DESTINATION_NAME_DEFAULT = null;
+    public static final String CAMEL_SOURCE_SJMS2_PATH_DESTINATION_NAME_CONF = "camel.source.path.destinationName";
+    public static final String CAMEL_SOURCE_SJMS2_PATH_DESTINATION_NAME_DOC = "DestinationName is a JMS queue or topic name. By default, the destinationName is interpreted as a queue name.";
+    
+    public CamelSjms2SourceConnectorConfig(ConfigDef config, Map<String, String> parsedConfig) {
+        super(config, parsedConfig);
+    }
+
+    public CamelSjms2SourceConnectorConfig(Map<String, String> parsedConfig) {
+        this(conf(), parsedConfig);
+    }
+
+    public static ConfigDef conf() {
+        ConfigDef conf = new ConfigDef(CamelSourceConnectorConfig.conf());
+
+        return conf.define(CAMEL_SOURCE_BRIDGE_ERROR_HANDLER_CONSUMER_CONF, ConfigDef.Type.BOOLEAN, CAMEL_SOURCE_BRIDGE_ERROR_HANDLER_CONSUMER_DEFAULT, ConfigDef.Importance.HIGH, CAMEL_SOURCE_BRIDGE_ERROR_HANDLER_CONSUMER_DOC)
+                .define(CAMEL_SOURCE_SJMS2_CONNECTION_FACTORY_CONF, ConfigDef.Type.STRING, CAMEL_SOURCE_SJMS2_CONNECTION_FACTORY_DEFAULT, ConfigDef.Importance.HIGH, CAMEL_SOURCE_SJMS2_CONNECTION_FACTORY_DOC)
```

Comment:
How the importance being determined here? I guess it depends if the property is required?

Reply:
Good observation, it wasn't determined. A criterion is good as any other I guess.

Reply:
No problems. We can iterate over it later since we will have this base. Thanks again for this !